velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -54,6 +54,19 @@ class ReaderBase {
     return FileMetaDataPtr(reinterpret_cast<const void*>(fileMetaData_.get()));
   }
 
+  const std::string& getNewFileSchemaColumnName(
+      uint32_t curSchemaIdx,
+      const std::string& name) const {
+    const auto& tableSchema = options_.fileSchema();
+    if (curSchemaIdx != 0) {
+      // this is column name at same index from table schema, we use will column
+      // name from table schema instead to build schemaWithId_ when
+      // we use column indices to map table columns to file columns
+      return tableSchema.get()->nameOf(curSchemaIdx - 1);
```

Comment:
@agrawalreetika Are you sure this mapping is correct? curSchemaIdx is the index of the schema element in the fileSchema, and all nested level would have a distinct curSchemaIdx. But the tableSchema is a RowType and indexed by column ids. They are not 1-1 mapping. I think you may want to try columnIdx instead.

Reply:
@agrawalreetika also the table schema needs to be recursively called. We saved one side recursion on file schema, but table schema also need to be traversed at the same time. For that we may need to add the table schema as a parameter to this function.